### PR TITLE
analytics(issue-views): Add analytics to issue views

### PR DESCRIPTION
**Pull request by MichaelSun48**
Adds analytics for the following events:

- User switched views
- User updated a view with a new query or sort 
- User discarded changes in a view 
- User renamed a view 
- User duplicated a view
- User deleted a view
- User reordered their views
- User clicked on `+ Add View`
- User saved a custom query from add view menu 
- User saved a deprecated saved search from add view menu 
- User saved a recommended view from add view menu 
- User opened a shared view from another user 
- User discarded a temporary view
- User saved a temporary view

### CODE REVIEW

File: static/app/views/issueList/groupSearchViewTabs/draggableTabBar.tsx [hunk 1]
```diff
@@ -267,6 +294,9 @@ export function DraggableTabBar({
       });
       setTabs(newTabs);
       tabListState?.setSelectedKey(tempId);
+      trackAnalytics('issue_views.add_view_clicked', {
```
Comment (roggenkemper) [inline static/app/views/issueList/groupSearchViewTabs/draggableTabBar.tsx:297]: i think some of these names are slightly different : `issue_views.add_view_clicked` vs `issue_views.add_view.clicked`

File: static/app/utils/analytics/issueAnalyticsEvents.tsx [hunk 2]
```diff
@@ -349,6 +373,21 @@ export const issueEventMap: Record<IssueEventKey, string | null> = {
   'issue_error_banner.proguard_misconfigured.clicked':
     'Proguard Potentially Misconfigured Issue Error Banner Link Clicked',
   'issues_tab.viewed': 'Viewed Issues Tab',
+  'issue_views.switched_views': 'Issue Views: Switched Views',
```
Comment (roggenkemper) [inline static/app/utils/analytics/issueAnalyticsEvents.tsx:376]: not sure how many of these are button clicks but if they are ,  we don't have to add an explicit call to `trackAnalytics` - https://develop.sentry.dev/development/analytics/#button-click-analytics

File: static/app/views/issueList/groupSearchViewTabs/draggableTabBar.tsx (hunk 1 above)
Comment (MichaelSun48) [reply static/app/views/issueList/groupSearchViewTabs/draggableTabBar.tsx:297]: ah my bad, I updated that in the definition and forgot to change it in the usages

File: static/app/utils/analytics/issueAnalyticsEvents.tsx (hunk 2 above)
Comment (MichaelSun48) [reply static/app/utils/analytics/issueAnalyticsEvents.tsx:376]: Good call - i there are definitely a couple places where we can replace the track analytics with this

File: static/app/views/issueList/overview.tsx [hunk 3]
```diff
@@ -197,6 +197,11 @@ class IssueListOverview extends Component<Props, State> {
     }
     this.fetchTags();
     this.fetchMemberList();
+    this.props.setRouteAnalyticsParams?.({
+      issueViewsEnabled: this.props.organization.features.includes(
```
Comment (roggenkemper) [inline static/app/views/issueList/overview.tsx:201]: we usually want to use snake case

**Review by roggenkemper — APPROVED** [review]

**Pull request merged**